unittests/snapshot_tests.cpp

```diff
@@ -384,4 +385,60 @@ BOOST_AUTO_TEST_CASE_TEMPLATE(test_compatible_versions, SNAPSHOT_SUITE, snapshot
    }
 }
 
+BOOST_AUTO_TEST_CASE_TEMPLATE(test_pending_schedule_snapshot, SNAPSHOT_SUITE, snapshot_suites)
+{
+   tester chain(setup_policy::preactivate_feature_and_new_bios);
+   auto block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 3); // ensure that test setup stays consistent with original snapshot setup
+   chain.create_account(N(snapshot));
+   block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 4);
+
+   const auto& gpo = chain.control->get_global_properties();
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.version, 0);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers.size(), 0);
+
+   auto res = chain.set_producers_legacy( {N(snapshot)} );
+   block = chain.produce_block();
+   BOOST_REQUIRE_EQUAL(block->block_num(), 5);
+   chain.control->abort_block();
+   ///< End deterministic code to generate blockchain for comparison
+   auto base_integrity_value = chain.control->calculate_integrity_hash();
+
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.version, 1);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers.size(), 1);
+   BOOST_REQUIRE_EQUAL(gpo.proposed_schedule.producers[0].producer_name.to_string(), "snapshot");
+
+   static_assert(chain_snapshot_header::minimum_compatible_version <= 2, "version 2 unit test is no longer needed.  Please clean up data files");
+   auto v2 = SNAPSHOT_SUITE::template load_from_file<snapshots::snap_v2_prod_sched>();
+   snapshotted_tester v2_tester(chain.get_config(), SNAPSHOT_SUITE::get_reader(v2), 0);
+   auto v2_integrity_value = v2_tester.control->calculate_integrity_hash();
+   if (std::is_same_v<SNAPSHOT_SUITE, variant_snapshot_suite> && v2_integrity_value.str() != base_integrity_value.str()) {
+      // create a latest snapshot
+      auto latest_writer = SNAPSHOT_SUITE::get_writer();
+      chain.control->write_snapshot(latest_writer);
+      auto chain_latest = SNAPSHOT_SUITE::finalize(latest_writer);
+      auto latest_writer2 = SNAPSHOT_SUITE::get_writer();
+      v2_tester.control->write_snapshot(latest_writer2);
+      auto v2_tester_latest = SNAPSHOT_SUITE::finalize(latest_writer2);
+      print_variant_diff(chain_latest, v2_tester_latest);
+      BOOST_REQUIRE_EQUAL(v2_integrity_value.str(), base_integrity_value.str());
```

Comment:
in the similar test `test_compatible_versions` we also validate a round-trip for the "v2" snapshot in that it can be loaded into controller, form it a new latest-version snapshot can be created and when that snapshot is loaded it still produces the same integrity value.   I'd like to see that round-trip here.

Reply:
I'll echo what @b1bart said here. I would also like to see the `v2_tester_latest` snapshot loaded and then compare the integrity hash of that to `v2_integrity_value`.

Reply:
@arhag I pushed up the change to make that happen.  Do you mind reviewing the whole of the PR and rendering a call?